spec.bs

```diff
@@ -320,26 +421,243 @@ maps to an internal [=fenced frame config=] [=struct=].
 
 Issue: Fix the "indistinguishable" IDL bug with the unions above.</span>
 
+<div algorithm>
+  In order to <dfn export>lift a [=urn uuid=] and [=fenced frame config=] into a
+  {{FencedFrameConfig}}</dfn>, run these steps:
+
+  1. <span class=XXX>TODO: Fill this in</span>
+</div>
+
 Each {{FencedFrameConfig}} has a <dfn for=fencedframeconfig>url</dfn>, which is a string, initially
-null.
+null. TODO: flesh out {{FencedFrameConfig}}
+
+A key feature of the <{fencedframe}> element is that web platform APIs can configure the behavior
+of the frame in a way that limits the ability of other execution contexts to modify or inspect this
+configuration, for security or privacy reasons. For example, the
+<a href=https://wicg.github.io/turtledove/>FLEDGE</a> API performs on-device ad auctions over
+cross-site data, and it is important that the ad that wins the auction can be loaded into a frame,
+without the API caller knowing *which ad* won the auction or being able to manipulate the
+environment in which the ad loads.
+
+We achieve this using the concept of a "fenced frame config". A fenced frame config is a collection
+of fields that can be loaded into <{fencedframe}> elements and that specifies the resulting
+environments. Fenced frame configs can only be constructed by web platform APIs, not initialized
+or modified arbitrarily. Their fields also contain "visibilities" for different entities, which
+dictate whether the field should be "redacted" when inspected by particular execution contexts.
+Each time a fenced frame config is loaded into a <{fencedframe}> element, it is instantiated as a
+new "fenced frame config instance", which governs that particular context.
+
+We now establish some preliminary types:
+
+A <dfn export for=fencedframeconfig>visibility</dfn> is either "<dfn export for=visibility>
```

Comment:
Not sure where this should go off the top of my head, but let's also mention somewhere in this section below that the creator of a config is responsible for passing in _everything_ that can go into it, and that there are no optional members, though they can be explicitly set to null. It could be a note, I'm just trying to make the flow clear somewhere. Maybe it would be good in the "Introduction"-style section that I mentioned above that we could add.

Reply:
Added this into the non-normative intro. Though if we are requiring [=fenced frame config=] to be constructed with explicit values for each field, maybe that should be somewhere normative.